packages/app-router/src/router.tsx

```diff
@@ -0,0 +1,335 @@
+  if (!Object.keys(layout.root).length) {
+    layout.root = {
+      stack: {
+        children: [
+          {
+            component: {
+              name: routes[0]?.name!,
+            },
+          },
+        ],
+      },
+    };
+  }
+
+  // Register the app launched listener to set the root layout
+  Navigation.events().registerAppLaunchedListener(async () => {
+    try {
+      await onAppLaunched?.();
+    } catch {
+      // handle the error (e.g., log it)
+    }
+
+    Navigation.setRoot(layout);
+  });
+}
+
+/**
+ * Adds a bottom tab to the root layout.
+ *
+ * @param {LayoutRoot} layout - The current layout object.
+ * @param {string} routeName - The name of the route associated with the bottom tab.
+ * @param {any} bottomTab - The bottom tab configuration.
+ *
+ * @example
+ * addBottomTabToLayout(layout, 'HomeScreen', { text: 'Home' });
+ */
+function addBottomTabToLayout(
+  layout: LayoutRoot,
+  routeName: string,
+  bottomTab: any,
+  stackId?: string,
+) {
+  const tab: LayoutStack = {
+    id: stackId,
+    children: [
+      {
+        component: {
+          name: routeName,
+        },
+      },
+    ],
+    options: {
+      bottomTab,
+    },
+  };
+
+  // Update the layout to include the new bottom tab
+  layout.root = {
+    bottomTabs: {
+      children: [...(layout.root.bottomTabs?.children ?? []), {stack: tab}],
+    },
+  };
+}
+
+/**
+ * Renders the component for a given route.
+ *
+ * @param {Route} route - The route definition.
+ * @param {any} props - The properties passed to the component.
+ * @param {React.ComponentType} Provider - The React component to be used as a provider.
+ * @param {Record<string, string>} routeMap - A map of route paths to route names.
+ * @param {React.ComponentType} ErrorBoundary - The component to be used as an error boundary.
+ * @returns {JSX.Element} The rendered component wrapped in the necessary providers.
+ *
+ * @example
+ * const renderedComponent = renderComponent(
+ *   { path: '/home', name: 'HomeScreen', Component: HomeComponent },
+ *   { componentId: 'component1', __flagship_app_router_url: 'https://example.com' },
+ *   MyCustomProvider,
+ *   routeMap,
+ *   MyErrorBoundary
+ * );
+ */
+function renderComponent(
+  route: ComponentRoute,
+  props: any,
+  Provider: React.ComponentType<any>,
+  ErrorBoundary: React.ComponentType<any>,
+  routes: Omit<RouteObject, 'children'>[],
+) {
+  const {componentId, APP_ROUTER_URL, ...data} = props;
+
+  // Ensure Component is defined (safeguard against potential undefined)
+  const Component = route.Component!;
+
+  // Memoize the URL if provided
+  const url: string | null = useMemo(() => {
+    if (!APP_ROUTER_URL) return null;
+
+    if (/^\S+:\/\//gm.test(APP_ROUTER_URL)) return APP_ROUTER_URL;
+
+    // TODO: use react-native-device-info to get bundleId / packageName as we use URL polyfill requiring standard URL
+    return `app://${APP_ROUTER_URL}`;
```

Comment:
likely no longer needed - we should be analyzing url within open function to determine if we should open link or path.